test/libyul/yulOptimizerAssemblyTests/commonSubexpressionEliminator/ethdebug_trivial.yul

```diff
@@ -0,0 +1,24 @@
+{
+    let a := /* @debug.set {"assignment":"a"} */ mul(1, codesize()) /* @debug.set {} */
+    let b := /* @debug.set {"assignment":"b"} */ mul(1, codesize()) /* @debug.set {} */
+}
```

Comment:
@ekpyron What result would you expect for this example? I would basically just merge the debug attributes here, basically creating a list of possible items:

```c++
// ----
// step: commonSubexpressionEliminator
//
// {
//     let a := /* @debug.set {"assignment": ["a", "b"]} */ mul(1, codesize()) /* @debug.set {} */
//     let b := /* @debug.set {"assignment": ["a", "b"]} */ a /* @debug.set {} */
// }
//
// Assembly:
//     /* "":58:68   */
//   codesize  // @debug.set {"assignment": ["a", "b"]}
//     /* "":55:56   */
//   0x01  // @debug.set {"assignment": ["a", "b"]}
//     /* "":51:69   */
//   mul  // @debug.set {"assignment": ["a", "b"]}
//     /* "":139:157   */
//   dup1  // @debug.set {"assignment": ["a", "b"]}
//     /* "":0:179   */
//   pop
//   pop
```

I think a debugger could easily find and use relevant information.

Reply:
However, I'm not very sure that this is correct in the general case.

Reply:
Maybe this would be more useful:
```c++
{
    let a := /* @debug.set {"assignment":"a", "data": 1} */ mul(1, codesize()) /* @debug.set {} */
    let b := /* @debug.set {"assignment":"b", "data": 2} */ mul(1, codesize()) /* @debug.set {} */
}
// ----
// step: commonSubexpressionEliminator
//
// {
//     let a := /* @debug.set [{"assignment":"a", "data": 1}, {"assignment":"b", "data": 2}] */ mul(1, codesize()) /* @debug.set {} */
//     let b := /* @debug.set [{"assignment":"a", "data": 1}, {"assignment":"b", "data": 2}] */ a /* @debug.set {} */
// }
//
// Assembly:
//     /* "":58:68   */
//   codesize  // @debug.set [{"assignment":"a", "data": 1}, {"assignment":"b", "data": 2}]
//     /* "":55:56   */
//   0x01  // @debug.set [{"assignment":"a", "data": 1}, {"assignment":"b", "data": 2}]
//     /* "":51:69   */
//   mul  // @debug.set [{"assignment":"a", "data": 1}, {"assignment":"b", "data": 2}]
//     /* "":139:157   */
//   dup1  // @debug.set [{"assignment":"a", "data": 1}, {"assignment":"b", "data": 2}]
//     /* "":0:179   */
//   pop
//   pop
```